src/WebJobs.Extensions.Http/ClaimsPrincipalBindingProvider.cs

```diff
@@ -0,0 +1,116 @@
+            "x-ms-functions-key-identity"
+        };
+
+        public Task<IBinding> TryCreateAsync(BindingProviderContext context)
+        {
+            if (context == null)
+            {
+                throw new ArgumentNullException("context");
+            }
+
+            if (context.Parameter.ParameterType != typeof(ClaimsPrincipal))
+            {
+                return NullBinding;
+            }
+
+            return Task.FromResult<IBinding>(new ClaimsIdentityBinding(context.Parameter));
+        }
+
+        private class ClaimsIdentityBinding : IBinding
+        {
+            private readonly ParameterInfo _parameter;
+
+            public ClaimsIdentityBinding(ParameterInfo parameter)
+            {
+                _parameter = parameter;
+            }
+
+            public bool FromAttribute
+            {
+                get { return false; }
+            }
+
+            public Task<IValueProvider> BindAsync(BindingContext context)
+            {
+                if (context == null)
+                {
+                    throw new ArgumentNullException("context");
+                }
+
+                IReadOnlyDictionary<string, object> bindingData = context.BindingData;
+                ClaimsPrincipal principal = GetClaimsPrincipalFromBindingData(bindingData);
+
+                var valueProvider = new SimpleValueProvider(typeof(ClaimsPrincipal), principal, principal?.Identity?.Name);
+                return Task.FromResult<IValueProvider>(valueProvider);
+            }
+
+            private static ClaimsPrincipal GetClaimsPrincipalFromBindingData(IReadOnlyDictionary<string, object> bindingData)
+            {
+                object headers;
+                if (bindingData.TryGetValue(HttpTriggerAttributeBindingProvider.HttpHeadersKey, out headers))
+                {
+                    List<ClaimsIdentity> identities = (headers as IDictionary<string, string>)
+                        .Where(header => PotentialIdentityHeaders.Any(idHeader => string.Equals(idHeader, header.Key, StringComparison.OrdinalIgnoreCase)))
+                        .Select(header => ClaimsIdentityHelper.FromBase64EncodedJson(header.Value))
+                        .ToList();
+                    return new ClaimsPrincipal(identities);
+                }
+                else
+                {
+                    return new ClaimsPrincipal();
+                }
+            }
+
+            public Task<IValueProvider> BindAsync(object value, ValueBindingContext context)
+            {
+                var request = value as ClaimsPrincipal;
+                if (request != null)
+                {
+                    var binding = new SimpleValueProvider(typeof(ClaimsPrincipal), request, "request");
+                    return Task.FromResult<IValueProvider>(binding);
+                }
+                throw new InvalidOperationException("Value must be of type " + typeof(ClaimsPrincipal).ToString());
```

Comment:
>Value [](start = 53, length = 5)

What does "Value" represent? Is it the name of a parameter? I suggest we make this exception message more precise to ensure the developer understands what they did wrong.

Reply:
I'm gonna take some additional inspriation from HttpTriggerAttributeBindingProvider and make this a NotSupportedException, and make the error message more clear.

---
In reply to: [181945049](https://github.com/Azure/azure-webjobs-sdk-extensions/pull/404#discussion_r181945049) [](ancestors = 181945049)